manim/animation/transform_matching_parts.py

```diff
@@ -153,8 +153,11 @@ def get_shape_map(self, mobject: Mobject) -> dict:
         return shape_map
 
     def clean_up_from_scene(self, scene: Scene) -> None:
+        # Interpolate all animations back to 0 to ensure source mobjects remain unchanged.
         for anim in self.animations:
-            anim.interpolate(0)
+            # Skip interpolating FadeIn because fade_target is not a source.
+            if not isinstance(anim, FadeIn):
+                anim.interpolate(0)
         scene.remove(self.mobject)
         scene.remove(self.to_remove)
```

Comment:
when doing a removal list instead of a single mobject
```suggestion
        scene.remove(*self.to_remove)
```